projects/github-actions/required-review/src/request-review.js

```diff
@@ -0,0 +1,66 @@
+const core = require( '@actions/core' );
+const github = require( '@actions/github' );
+const { WError } = require( 'error' );
+
+/**
+ * Fetch Username based on @user
+ *
+ * @param {string} user - @ followed by a GitHub user name.
+ */
+async function getUsername( user ) {
+    const octokit = github.getOctokit( core.getInput( 'token', { required: true } ) );
+    try {
+        const res = await octokit.rest.users.getByUsername( { username: user.slice( 1 ) } );
+        return res.data.login
+    } catch ( error ) {
+        throw new WError(
+            // prettier-ignore
+            `Failed to query user ${ user } from GitHub: ${ error.response?.data?.message || error.message }`,
+            error,
+            {}
+        );
+    }
+}
+
+/**
+ * Request review from the given team
+ *
+ * @param {string} team - GitHub team slug, or @ followed by a GitHub user name.
+ */
+async function requestReviewer( team ) {
+	const octokit = github.getOctokit( core.getInput( 'token', { required: true } ) );
+	const owner = github.context.payload.repository.owner.login;
+	const repo = github.context.payload.repository.name;
+	const pr = github.context.payload.pull_request.number;
+
+    if ( team.startsWith( '@' ) ) {
+		// Handle @singleuser virtual teams. Fetch the correct username case from GitHub
+		// to avoid having to worry about edge cases and Unicode versions and such.
+        try {
+            const login = await getUsername(team);
```

Comment:
I like that you extracted that here, that's clean. 👍

I wonder if we should make further use of that extraction in the existing logic in `projects/github-actions/required-review/src/team-members.js`?

Reply:
yup, I can pull this out. Is there a file or location that is preferred for this? ie a helper file, or I can call it `get-username.js` and reference it in both files.

Reply:
> or I can call it get-username.js and reference it in both files

This seems like a good approach to me 👍